packages/composite-checkout-wpcom/src/components/form-field-annotation.tsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent } from 'react';
+import styled from '@emotion/styled';
+
+type FormFieldAnnotationProps = {
+	// Semantic props
+	labelText: string;
+	descriptionText?: string;
+	errorMessage: string;
+
+	// Functional props
+	isError: boolean;
+	isDisabled: boolean;
+
+	// Technical props
+	className?: string;
+	labelId: string;
+	descriptionId: string;
+	formFieldId: string;
+};
+
+export const FormFieldAnnotation: FunctionComponent< FormFieldAnnotationProps > = ( {
+	formFieldId,
+	labelText,
+	labelId,
+	descriptionText,
+	descriptionId,
+	errorMessage,
+	isError,
+	isDisabled,
+	className,
+	children,
+} ) => (
+	<div className={ className }>
+		{ labelText && (
+			<Label htmlFor={ formFieldId } id={ labelId } isDisabled={ isDisabled }>
+				{ labelText }
+			</Label>
+		) }
+		<FormFieldWrapper data-testid={ `${ className }_wrapper` } isError={ isError }>
+			{ children }
+		</FormFieldWrapper>
+		<RenderedDescription
+			descriptionText={ descriptionText }
+			descriptionId={ descriptionId }
+			isError={ isError }
+			errorMessage={ errorMessage }
+		/>
+	</div>
+);
+
+type LabelProps = {
+	isDisabled: boolean;
+	theme?: any;
+};
+
+const Label = styled.label< LabelProps >`
+	display: block;
+	color: ${props => props.theme.colors.textColor};
+	font-weight: ${props => props.theme.weights.bold};
+	font-size: 14px;
+	margin-bottom: 8px;
+
+	:hover {
+		cursor: ${props => ( props.isDisabled ? 'default' : 'pointer' )};
+	}
+`;
+
+type FormFieldWrapperProps = {
+	isError: boolean;
+	theme?: any;
+};
+
+const FormFieldWrapper = styled.div< FormFieldWrapperProps >`
+	display: block;
+	width: 100%;
+	box-sizing: border-box;
+	font-size: 16px;
+	border: 1px solid
+		${props => ( props.isError ? props.theme.colors.error : props.theme.colors.borderColor )};
```

Comment:
It's interesting that Prettier formatted it this way -- there should be a space: `${ props => ( ... ) }`